css-view-transitions-2/two-phase-transition-explainer.md

```diff
@@ -0,0 +1,54 @@
+# Two-phase view transition
+Making cross-document view-transitions feel instant and seamless.
+
+See w3c/csswg-drafts#10616
+
+# Current state
+View-transitions don’t start immediately - rendering is paused until the new state is ready. 
+This is especially visible in cross-document view transitions, where the transition only captures the old when the new document’s response arrives, and starts when the new document is ready to render.
+
+This can create a jarring or abrupt experience.
+
+# Current workarounds
+
+## Starting an animation manually on the old page
+This is possible, however the old page cannot delay the navigation, which means the animation is likely to be interrupted when the navigation commits.
+
+## Intercepting the navigation and restarting it when the animation ends
+This works but is hard to achieve without slowing down the whole navigation process.
+
+## Timing out render-blocking
+This can reduce the “frozen” time, however it means the transition doesn’t end at the optimal state, and also doesn’t help with making it feel instant.
+
+# Proposed solution
+The core of the problem is that view-transitions require a start and an end phase before starting, but we don’t know the end phase in advance when it is computed in an async function (e.g. in a navigation).
+
+## In a nutshell
+Proposing to prototype a “two-phase” view-transition: 
+Instantly start a transition to a state that can be computed quickly enough or synchronously. Call this a “preview” state.
+When the to-preview transition ends, transition from there to the final state.
+Only interrupt the first transition after a timeout, otherwise stall the navigation commit until finished.
+This should ideally not delay the LCP / loading experience of the new page, as the content keeps loading (and potentially prerendering) in the background.
+
+## Phase 1: script-invoked preview
+We can create this kind of seamless/instant experience without any new CSS, and potentially without needing to fully spec it normatively by changing the behavior as follows:
```

Comment:
I'm worried that we might be blocked by some unrelated view transition as a result of this (especially some long running one). I wonder if we can explicitly mark one as a preview transition via types or something

Reply:
> I'm worried that we might be blocked by some unrelated view transition as a result of this (especially some long running one). I wonder if we can explicitly mark one as a preview transition via types or something

Sure or call navigateEvent.startViewTransition or so